packages/@aws-cdk/aws-s3-assets/lib/asset.ts

```diff
@@ -198,7 +198,17 @@ export class Asset extends cdk.Construct implements cdk.IAsset {
   }
 }
 
-function determinePackaging(assetPath: string): cdk.FileAssetPackaging {
+/**
+ * Determine the type of packaging to use for the S3 asset.
+ *
+ * @param copyFromImagePath When true, the asset will be a directory copied
+ * from the bundler image, so the packaging should be a ZIP_DIRECTORY.
+ */
+function determinePackaging(assetPath: string, copyFromImagePath?: boolean): cdk.FileAssetPackaging {
+  if (copyFromImagePath) {
+    return cdk.FileAssetPackaging.ZIP_DIRECTORY;
+  }
```

Comment:
This is a "trick" that shows that `path` has no meaning anymore here and to avoid checking its existence we need to return early.